packages/lesswrong/lib/collections/comments/views.ts

```diff
@@ -29,20 +29,29 @@ Comments.addDefaultView((terms: CommentsViewTerms, _, context?: ResolverContext)
   const validFields = pick(terms, 'userId', 'authorIsUnreviewed');
 
   const alignmentForum = forumTypeSetting.get() === 'AlignmentForum' ? {af: true} : {}
+  const hideSince = hideUnreviewedAuthorCommentsSettings.get()
+  // When we're hiding unreviewed comments, we allow comments that meet any of:
+  //  * The author is reviewed
+  //  * The comment was posted before the hideSince date
+  //  * The comment was posted by the current user
+  // Do not run this on the client (ie: Mingo), because we will not have the
+  // resolver context, and return unreviewed comments to ensure cache is
+  // properly invalidated.
   // We set `{$ne: true}` instead of `false` to allow for comments that haven't
   // had the default value set yet (ie: those created by the frontend
   // immediately prior to appearing)
-  const hideUnreviewedAuthorComments = hideUnreviewedAuthorCommentsSettings.get()
+  const hideNewUnreviewedAuthorComments = (hideSince && bundleIsServer)
```

Comment:
It's a bit hard to reason about what happens when this view has a difference between server and mingo. The one case that I expect might break is, when an unreviewed author posts a comment, does that comment appear on the page for them without a refresh? If that's tested and still works then I think this is fine.

Reply:
That's correct, it used to not do that, because context was only defined on the server, and so in the client it would exclude the comment. On the client, we can be too-inclusive. The only comments being checked should be the authors, and in any case, we would only be causing a refresh.